paddle/fluid/operators/one_hot_op.cc

```diff
@@ -52,11 +63,22 @@ class OneHotOpMaker : public framework::OpProtoAndCheckerMaker {
              "(LoDTensor, LoDTensor<int>) Input variable with rank at least 2. "
              "The last dimension of X should be 1. Each value of X is an index "
              "to indicate the position.");
+    AddInput("depth", "(Tensor, Tensor<int>), Length of one-hot vector")
+        .SetType("int32")
+        .RemainCPU()
+        .OnlyInStaticModel()
+        .AsDispensable();
     AddOutput("Out",
               "(Tensor, Tensor<float>) Output tensor with same rank as X. "
               "The tensor consists of one-hot representations of values in X.");
-    AddAttr<int>("depth",
-                 "A positive integer to specify the length of one-hot vector.");
+
+    AddAttr<int>("depth_attr",
+                 "A positive integer to specify the length of one-hot vector.")
+        .SetDefault(-1);
+    AddAttr<bool>(
+        "use_attr",
+        "An boolen to special to use depth in attr or use depth in Input")
+        .SetDefault(false);
```

Comment:
Why do you change the attributes and input? you should consider the compatibility.